Simbody/src/CableSpan.cpp

```diff
@@ -0,0 +1,3028 @@
+                    decorations.push_back(
+                        DecorativeLine(prevPoint_G, nextPoint_G)
+                            .setColor(c_Color)
+                            .setLineThickness(c_LineThickness));
+                }
+                prevPoint_G  = nextPoint_G;
+                isFirstPoint = false;
+            };
+            // Call the lambda for all path points.
+            calcDecorativePathPoints(
+                state,
+                pushDecorativeLineBetweenPathPoints);
+        }
+
+        // For any obstacle that is not in contact with the cable, draw the
+        // tracking point used for touchdown.
+        {
+            const Vec3 c_Color            = Red;
+            constexpr int c_LineThickness = 3;
+
+            // Find those obstacles that are not in contact.
+            for (ObstacleIndex ix(0); ix < getNumObstacles(); ++ix) {
+                const CurveSegment& curve = getObstacleCurveSegment(ix);
+                if (curve.isInContactWithSurface(state)) {
+                    continue;
+                }
+
+                // Draw the tracking point (in ground frame).
+                const Transform X_GS = curve.getDataPos(state).X_GS;
+                const Vec3 trackingPoint_S =
+                    curve.getDataInst(state).trackingPointOnLine_S;
+                const Vec3 trackingPoint_G =
+                    X_GS.shiftFrameStationToBase(trackingPoint_S);
+                decorations.push_back(
+                    DecorativePoint(trackingPoint_G).setColor(c_Color));
+                // Draw line to tracking point from either the surface, or the
+                // obstacle origin. We default to drawing from the origin
+                // because surface projection becomes unstable at larger
+                // distances, which causes the projection to take too much
+                // time. Which was the reason to not use it in the first place.
+                // But it is more illustrative.
+                constexpr bool c_DrawLineFromObstacleOrigin = true;
+                if (c_DrawLineFromObstacleOrigin) {
+                    decorations.push_back(
+                        DecorativeLine(X_GS.p(), trackingPoint_G)
+                            .setColor(c_Color)
+                            .setLineThickness(c_LineThickness));
+                } else {
+                    try {
+                        const Vec3 surfacePoint_S =
+                            curve.getContactGeometry()
+                                .projectDownhillToNearestPoint(trackingPoint_S);
+                        const Vec3 surfacePoint_G =
+                            X_GS.shiftFrameStationToBase(surfacePoint_S);
+                        decorations.push_back(
+                            DecorativeLine(trackingPoint_G, surfacePoint_G)
+                                .setColor(Orange)
+                                .setLineThickness(c_LineThickness));
+                    } catch (...) {
```

Comment:
Gotta catch them alll.....

Reply:
changed to:
```cpp
                    try {
                        surfacePoint_S =
                            curve.getContactGeometry()
                                .projectDownhillToNearestPoint(trackingPoint_S);
                    } catch (...) {
                        // Yes catching all is bad, but since this plotting
                        // option is just for debugging, and this
                        // projection is likely to fail as we get further
                        // from the surface, we just dont plot it if we
                        // don't have it.
                        continue;
                    }
```